### PR TITLE
Dynamic downloader

### PR DESCRIPTION
**Pull request by Gaojianli**
## Summary by CodeRabbit

- **New Features**
  - Added the ability to edit and delete clients directly from the dashboard.
  - Introduced a modal for creating or updating downloader configurations, including support for qBittorrent and Transmission.

- **Enhancements**
  - Implemented success and error messages for downloader operations.
  - Added fade-in effects for edit buttons on hover for better user experience.

### CODE REVIEW

File: src/views/dashboard/components/clientStatus.vue [hunk 1]
```diff
@@ -1,5 +1,18 @@
 <template>
-  <a-typography-title :heading="3">{{ t('page.dashboard.clientStatus.title') }}</a-typography-title>
+  <a-typography-title :heading="3"
+    >{{ t('page.dashboard.clientStatus.title') }}
+    <a-button
+      class="add-btn"
+      type="outline"
+      :shape="'circle'"
```
Comment (XiaoXice) [inline src/views/dashboard/components/clientStatus.vue]: 这里可以节约三个字符（

File: src/views/dashboard/components/clientStatus.vue [hunk 2]
```diff
@@ -36,17 +49,21 @@
       <ClientStatusCard
         :downloader="client"
         @torrent-view-click="() => torrentList?.showModal(client.name)"
+        @downloader-delete="refresh"
```
Comment (XiaoXice) [inline src/views/dashboard/components/clientStatus.vue]: 组件名字已经叫 ClientStatusCard 了，不如改名为 after-delete 或者 delete-success

File: src/views/dashboard/components/clientStatus.vue [hunk 3]
```diff
@@ -36,17 +49,21 @@
...
+        @edit-click="(e) => editDownloaderModal?.showModal(false, e)"
       />
     </a-col>
   </a-row>
   <TorrentListModal ref="torrentList" />
+  <EditDownloaderModal ref="editDownloaderModal" @downloader-changed="refresh" />
```
Comment (XiaoXice) [inline src/views/dashboard/components/clientStatus.vue]: downloader-changed 更名 changed

File: src/views/dashboard/components/clientStatusCard.vue [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+          @click="() => emits('edit-click', { name: downloader.name, config: client?.config!! })"
+        >
+          <template #icon>
+            <icon-edit />
+          </template>
+        </a-button>
+        <a-popconfirm
+          :content="t('page.ruleSubscribe.column.deleteConfirm')"
+          type="warning"
+          @before-ok="() => handleDelete(downloader)"
```
[the remark this answers is not in the document]
Comment (XiaoXice) [reply src/views/dashboard/components/clientStatusCard.vue]: 我觉得说得对
downloader 已经在 props 里了，没必要再包装一个箭头函数了。

**Review by XiaoXice — APPROVED** [review]
有些命名可改可不改。看你心情。

File: src/views/dashboard/components/clientStatusCard.vue [hunk 5]
```diff
@@ -1,5 +1,30 @@
 <template>
-  <a-card hoverable style="height: 100%" :header-style="{ height: 'auto' }">
+  <a-card hoverable style="height: 100%" :header-style="{ height: 'auto' }" class="card">
+    <template #extra>
+      <a-space v-if="client" size="mini">
+        <a-button
+          class="edit-btn"
+          shape="circle"
+          type="text"
+          @click="() => emits('edit-click', { name: downloader.name, config: client?.config!! })"
```
[the remark this answers is not in the document]
Comment (Gaojianli) [reply src/views/dashboard/components/clientStatusCard.vue:9]: 为什么会改善性能？

**Pull request merged**